plugins/scaffolder-node/report-alpha.api.md

```diff
@@ -27,12 +50,307 @@ export type AutocompleteHandler = ({
+  permission: ResourcePermission<'scaffolder-template'>,
+  conditions: PermissionCriteria<PermissionCondition<'scaffolder-template'>>,
+) => ConditionalPolicyDecision;
+
+// @public (undocumented)
+export const createTemplatePermissionRule: <
+  TParams extends PermissionRuleParams = undefined,
+>(
+  rule: PermissionRule<
+    TemplateParametersV1beta3 | TemplateEntityStepV1beta3,
+    {},
+    'scaffolder-template',
+    TParams
+  >,
+) => PermissionRule<
+  TemplateParametersV1beta3 | TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-template',
+  TParams
+>;
+
+// @public (undocumented)
+export const hasAction: PermissionRule<
+  TemplateEntityV1beta3,
+  {},
+  'scaffolder-entity',
+  {
+    actionId: string;
+  }
+>;
+
+// @public (undocumented)
+export const hasActionId: PermissionRule<
+  TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-action',
+  {
+    actionId: string;
+  }
+>;
+
+// @public (undocumented)
+export const hasActionWithBooleanProperty: PermissionRule<
+  TemplateEntityV1beta3,
+  {},
+  'scaffolder-entity',
+  {
+    actionId: string;
+    key: string;
+    value?: boolean | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasActionWithNumberProperty: PermissionRule<
+  TemplateEntityV1beta3,
+  {},
+  'scaffolder-entity',
+  {
+    actionId: string;
+    key: string;
+    value?: number | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasActionWithProperty: PermissionRule<
+  TemplateEntityV1beta3,
+  {},
+  'scaffolder-entity',
+  {
+    actionId: string;
+    key: string;
+    value?: string | number | boolean | null | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasActionWithStringProperty: PermissionRule<
+  TemplateEntityV1beta3,
+  {},
+  'scaffolder-entity',
+  {
+    actionId: string;
+    key: string;
+    value?: string | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasBooleanProperty: PermissionRule<
+  TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-action',
+  {
+    key: string;
+    value?: boolean | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasNumberProperty: PermissionRule<
+  TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-action',
+  {
+    key: string;
+    value?: number | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasProperty: PermissionRule<
+  TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-action',
+  {
+    key: string;
+    value?: string | number | boolean | null | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasStringProperty: PermissionRule<
+  TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-action',
+  {
+    key: string;
+    value?: string | undefined;
+  }
+>;
+
+// @public (undocumented)
+export const hasTag: PermissionRule<
+  TemplateParametersV1beta3 | TemplateEntityStepV1beta3,
+  {},
+  'scaffolder-template',
+  {
+    tag: string;
+  }
+>;
+
+// @public (undocumented)
+export const hasTaggedAction: PermissionRule<
```

Comment:
Don't think we should export these permission rules directly, use the namespace below.